src/ansys/aedt/toolkits/antenna/run_toolkit.py

```diff
@@ -1,161 +1,85 @@
-                print(f"Killing process {process.pid} on {ip}:{port}")
-                process.terminate()
-
-
-def check_backend_communication():
-    response = requests.get(URL_CALL + "/health")
-    if response.ok:
-        return True
-    return False
-
-
-def process_desktop_properties():
-    desktop_pid = None
-    desktop_version = None
-    grpc = False
-    if "PYAEDT_SCRIPT_VERSION" in os.environ and (
-        "PYAEDT_SCRIPT_PROCESS_ID" in os.environ or "PYAEDT_SCRIPT_PORT" in os.environ
-    ):
-        desktop_version = os.environ["PYAEDT_SCRIPT_VERSION"]
-        desktop_pid = os.environ[
-            "PYAEDT_SCRIPT_PORT" if desktop_version > "2023.2" or IS_LINUX else "PYAEDT_SCRIPT_PROCESS_ID"
-        ]
-        grpc = desktop_version > "2023.2" or IS_LINUX
-
-    elif len(sys.argv) == 3:
-        desktop_pid, desktop_version = sys.argv[1], sys.argv[2]
-
-    if desktop_pid and desktop_version:
-        properties = {
-            "selected_process": int(desktop_pid),
-            "aedt_version": desktop_version,
-            "use_grpc": grpc,
-        }
-        requests.put(URL_CALL + "/set_properties", json=properties)
-        # requests.post(URL_CALL + "/launch_aedt")
-
+backend_file = os.path.join(backend.__path__[0], "run_backend.py")
+frontend_file = os.path.join(ui.__path__[0], "run_frontend.py")
+
+is_linux = os.name == "posix"
+new_port = find_free_port(backend_properties.url, backend_properties.port)
+if not new_port:
+    raise Exception("No free ports in {}".format(backend_properties.url))
+
+frontend_properties.backend_port = new_port
+url = frontend_properties.backend_url
+port = frontend_properties.backend_port
+python_path = sys.executable
+backend_command = [python_path, backend_file, url, str(port)]
+frontend_command = [python_path, frontend_file, url, str(port)]
+url_call = f"http://{url}:{port}"
 
 # Main Execution
 
 # Clean python process when script ends
-atexit.register(clean_python_processes)
+atexit.register(clean_python_processes, url, port)
 
 # Check if backend is already running
-is_server_busy = is_server_running(server=URL, port=PORT)
+is_server_busy = is_server_running(server=url, port=port)
 if is_server_busy:
-    KILL_BACKEND = False
-    raise Exception("There is a process running in: {}".format(URL_CALL))
+    raise Exception("There is a process running in: {}".format(url_call))
 
 # Launch backend thread
-backend_thread = server_actions(BACKEND_COMMAND, "template_backend")
+backend_thread = server_actions(backend_command, "template_backend", is_linux)
 
 # Connect to AEDT session if arguments or environment variables are passed
-process_desktop_properties()
+process_desktop_properties(is_linux, url_call)
 
 # Launch frontend thread
-frontend_thread = server_actions(FRONTEND_COMMAND, "template_frontend")
+frontend_thread = server_actions(frontend_command, "template_frontend", is_linux)
 
 # Check if backend is running. Try every 1 second with a timeout of 10 seconds
```

Comment:
```suggestion
# Check if backend is running. Try every 1 second with a timeout of 10 seconds.
```
Add a period here because there are two sentences (and the first has a period).